README.md

```diff
@@ -233,32 +258,84 @@ Following are about the toggles:
+4. Run your app (e.g. using iOS simulator) via `flutter run integration_test/main_test.dart --host-vmservice-port 9753 --disable-service-auth-codes --dart-define "CONVENIENT_TEST_APP_CODE_DIR=$PWD"`. Can also be run via VSCode or Android Studio with similar commands.
    - for use with the Android emulator, add `--dart-define CONVENIENT_TEST_MANAGER_HOST=10.0.2.2`
-4. Run the GUI located in `packages/convenient_test_manager`. (See details in section below)
-5. Enjoy the GUI!
+5. Run the Convenient Test Manager (See details in section below)
 
 ### Run the `manager`
 
+- [`packages/convenient_test_manager`](packages/convenient_test_manager) is a flutter GUI app that allows running and inspecting individual tests.
+  Use this for development and debugging.
+- [`packages/convenient_test_manager_dart`](packages/convenient_test_manager_dart) is a dart CLI app that runs all tests and terminates with a corresponding exit code.
+  Use this for CI and automated testing. See [CI / headless mode](#ci--headless-mode)
+
 #### Method 1: Download precompiled binary
 
-There are precompiled binaries for each commit, at the "artifacts" section in [the CI page](https://github.com/fzyzcjy/flutter_convenient_test/actions/workflows/ci.yaml). For example, download `manager_macos` artifact, unzip it, and open it.
+See [Releases](https://github.com/fzyzcjy/flutter_convenient_test/releases) for binaries.
+
+Additionally, there are precompiled binaries for each commit, at the "artifacts" section in [the CI page](https://github.com/fzyzcjy/flutter_convenient_test/actions/workflows/ci.yaml)
+(download only available for logged in GitHub users).
+For example, go to the most recent _CI_ run, download `manager_gui_macos` artifact, unzip it, and open it.
+
+#### Method 2: `flutter run` / `flutter build`
+
+Check the requirements for developing flutter desktop apps for
+[macOS](https://docs.flutter.dev/platform-integration/macos/install-macos/install-macos-from-ios),
+[Linux](https://docs.flutter.dev/platform-integration/linux/install-linux/install-linux-from-android),
+and [Windows](https://docs.flutter.dev/platform-integration/windows/install-windows/install-windows-from-android).
+
+```sh
+git clone https://github.com/fzyzcjy/flutter_convenient_test.git
+cd flutter_convenient_test/packages/convenient_test_manager
 
-Currently only MacOS and Linux has CI for this, and I am willing to accept a PR for Windows.
+# compile executable. platform = linux | macos | windows
+flutter build <platform>
+# or just run directly
+flutter run -d <platform>
+```
+
+##### For CLI manager:
+
+```sh
+cd flutter_convenient_test/packages/convenient_test_manager
+
+dart compile exe bin/convenient_test_manager_dart.dart
+# or just run directly
+dart run
+```
+
+#### Method 3: `dart pub global` (only CLI manager)
```

Comment:
nit: maybe add a sentence telling users, this is the standard and normal way to run a dart program

Reply:
if i understand this correctly, you mean this in the same way as the above comment for the gui manager. if this is the case, then this is [done](https://github.com/fzyzcjy/flutter_convenient_test/pull/399/commits/e3872c82b0b5f1368d46a5dbd0f2f4ea5b9a0935), same as for the above comment.